src/test/java/org/assertj/core/error/ShouldHaveDuration_create_test.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2019 the original author or authors.
+ */
+package org.assertj.core.error;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.error.ShouldHaveDuration.shouldHaveDays;
+import static org.assertj.core.error.ShouldHaveDuration.shouldHaveHours;
+import static org.assertj.core.error.ShouldHaveDuration.shouldHaveMillis;
+import static org.assertj.core.error.ShouldHaveDuration.shouldHaveMinutes;
+import static org.assertj.core.error.ShouldHaveDuration.shouldHaveNanos;
+import static org.assertj.core.error.ShouldHaveDuration.shouldHaveSeconds;
+
+import java.time.Duration;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author Filip Hrisafov
+ */
+class ShouldHaveDuration_create_test {
+
+  @Test
+  void should_create_error_message_for_nanos() {
+    // GIVEN
+    Duration duration = Duration.ofNanos(1893);
+    long actualNanos = 1893;
+    long expectedNanos = 190;
+    // WHEN
+    String errorMessage = shouldHaveNanos(duration, actualNanos, expectedNanos).create();
+    // THEN
+    assertThat(errorMessage).isEqualTo(format("%nExpecting Duration:%n <PT0.000001893S>%nto have 190L nanos but had 1893L"));
```

Comment:
Minor about style: `then()` can be used instead of `assertThat()`

Reply:
I used another test as a reference. And honestly I prefer writing less code. Feel free to change the tests